core/src/test/java/test/retryAnalyzer/dataprovider/RetryAnalyzerWithComplexDataProviderTest.java

```diff
@@ -10,13 +10,52 @@
 public class RetryAnalyzerWithComplexDataProviderTest extends SimpleBaseTest {
 
     @Test(description = "GITHUB-2148")
-    public void ensureTestsAreRetried() {
-        TestNG testng = create(ComplexDataProviderSample.class);
+    public void testWithoutDataProvider() {
+        TestNG testng = create(RetryAnalyzerWithoutDataProvider.class);
+        TestListenerAdapter tla = new TestListenerAdapter();
+        testng.addListener(tla);
+        testng.run();
+        assertEquals(tla.getPassedTests().size(), 1);
+        assertEquals(tla.getSkippedTests().size(), 3);
+    }
+
+    @Test(description = "GITHUB-2148")
+    public void testWithDataProviderStringArray() {
+        TestNG testng = create(DataProviderWithStringArraySample.class);
+        TestListenerAdapter tla = new TestListenerAdapter();
+        testng.addListener(tla);
+        testng.run();
+        assertEquals(tla.getPassedTests().size(), 1);
+        assertEquals(tla.getSkippedTests().size(), 3);
+    }
+
+    @Test(description = "GITHUB-2148")
+    public void testWithSingleParam() {
+        TestNG testng = create(DataProviderWithSingleParam.class);
+        TestListenerAdapter tla = new TestListenerAdapter();
+        testng.addListener(tla);
+        testng.run();
+        assertEquals(tla.getPassedTests().size(), 2);
```

Comment:
I supposed that all test cases were testing the same things but with different kinds of params.
Now, it is not obvious to me why this test case is the only one with 2 passed tests.

Reply:
Added a comment in each test-cases to make it clear. Since we are passing to integer array of objects the number of test cases count is 2 and this is existing testng behaviour.